final_task/rss_reader/rss_parser.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3.8
+
+import html
+from os import path
+
+from bs4 import BeautifulSoup
+import feedparser
+from jinja2 import Environment, FileSystemLoader
+import requests
+
+from rss_reader import rss_exceptions
+
+
+class RSSparser:
+    """
+    Parsed the RSS news from received link, extracts required amount of news and prints result
+    in human readable format.
+
+    Parameters:
+        args: dict: parsed arguments out of sys.argv;
+        logger: LOGGER: tracks events that occur during program execution.
+
+    Returns:
+        store_news: list of dictionaries with extracted info of required amount of parsed news;
+        output_txt_news method returns result in human readable format (filling the prepared).
+    """
+
+    def __init__(self, args, logger):
+        self.url = args.source
+        self.limit = args.limit
+        self.logger = logger
+
+        self.logger.info(f'Get RSS_url {self.url} and value = {self.limit} (limits amount of output news)')
+
+        self.check_the_connection(self.url, self.logger)
+        self.response = self.get_the_response(self.url, self.logger)
+
+        # Extract the news-site name converting it to the unicode
+        self.feed_name = html.unescape(self.response.feed.get('title', ''))
+        # Extract all news separately in one list
+
+        self.logger.info('Separate a news from the URL.')
+
+        self.news = self.response.entries
+
+        self.limit = self.check_limit_value(self.limit)
+        self.all_news = self.parse_feed()
+
+    @staticmethod
+    def check_the_connection(url, logger):
+        """
+        Check the internet connection
+        :param url: str
+        :param logger: logger
+        :return: None
+        """
+        try:
+            requests.get(url, timeout=1)
+            logger.info('Check the Internet connection.')
+        except (
+                requests.exceptions.ConnectionError,
+                requests.exceptions.Timeout,
+                requests.exceptions.HTTPError,
+        ) as error:
+            raise SystemExit(f'Error: {error}. URL {url} is unreachable.')
+        else:
+            logger.info('Connection established.')
+
+    @staticmethod
+    def check_response_status_code(response, logger):
+        """
+        Check if the response status code is 200: OK
+        :param response: dict
+        :param logger: logger
+        :return: None
+        """
+        if response.get('status', '') != 200:
+            raise Exception(f'Bad response status code {str(response.status)}')
```

Comment:
It is better to use `response.raise_for_status` here

Reply:
I used feedparser.parse(url) for getting the response, so therefor I can't implement  response.raise_for_status

Reply:
Have you tried this?

Reply:
yes and it doesn't work

Reply:
Ok. Then please be sure check some methods that this `response` object has. I am pretty sure that there must be something for this. However if there is nothing there, then leave it as is.